### PR TITLE
55 unhappy code tests

### LINKED ISSUE #55: Tests should cover unhappy code paths in configuration change parsing
https://codecov.io/gh/ess-dmsc/forwarder/src/afbb5c66f457962596f3ffb421e0bb9eeea11c89/forwarder/parse_config_update.py

### PR DESCRIPTION
**Pull request by kmurica**
Closes #55

### CODE REVIEW

File: tests/parse_config_update_test.py [hunk 1]
```diff
@@ -170,3 +173,53 @@ def test_parse_streams_skips_stream_info_if_remove_config_and_schema_present_but
...
+    message = serialise_rf5k(
+        invalid_update_type,
+        [StreamInfo(test_channel_name, "f142", "output_topic", Protocol.PVA)],
+    )
+    config_update = parse_config_update(message)
+    assert config_update.command_type == CommandType.INVALID
+
+
+def test_parse_streams_when_invalid_protocol_is_provided():
+    invalid_protocol = Protocol.FAKE + 1
```
Comment (mattclarke) [inline tests/parse_config_update_test.py:218]: This test will start failing if we add a new protocol and suddenly Protocol.FAKE + 1 is valid.
Same for some of the other tests above.

Perhaps we can use negative values instead, if that is allowed by the underlying library.
If that is not possible then a much bigger number will have to do.

File: tests/parse_config_update_test.py [hunk 2]
```diff
@@ -170,3 +173,53 @@ def test_parse_streams_skips_stream_info_if_remove_config_and_schema_present_but
...
     streams = tuple(_parse_streams(CommandType.REMOVE, config_message.streams))
     assert not streams
+
+
+@patch(
+    "forwarder.parse_config_update.deserialise_rf5k",
+    side_effect=RuntimeError("Runtime Error"),
+)
+def test_parse_config_update_when_runtime_error_occurs(mock_func):
+    test_channel_name = "test_channel"
```
Comment (mattclarke) [inline tests/parse_config_update_test.py:183]: This doesn't need to a variable. The string can be put directly into StreamInfo

File: tests/parse_config_update_test.py [hunk 3]
```diff
@@ -170,3 +173,53 @@ def test_parse_streams_skips_stream_info_if_remove_config_and_schema_present_but
...
+    )
+    config_update = parse_config_update(message)
+    assert config_update.command_type == CommandType.INVALID
+
+
+@patch(
+    "forwarder.parse_config_update.deserialise_rf5k",
+    side_effect=flatbuffer_struct.error("Flatbuffer Error"),
+)
+def test_parse_config_update_when_flatbuffer_struct_error_occurs(mock_func):
```
Comment (mattclarke) [inline tests/parse_config_update_test.py:196]: These tests need better names as they should act as documentation of the system.
Perhaps something like "command is invalid on flatbuffer struct error"

File: tests/parse_config_update_test.py (hunk 2 above)
Comment (kmurica) [reply tests/parse_config_update_test.py]: Yeah, never copy-paste code. ;) Fixed.

File: tests/parse_config_update_test.py (hunk 3 above)
Comment (kmurica) [reply tests/parse_config_update_test.py]: Can you check if the names are ok now?

File: tests/parse_config_update_test.py (hunk 1 above)
Comment (kmurica) [reply tests/parse_config_update_test.py]: Negative are not possible so I put 9999 (clear indication hopefully that it is invalid and should be large enough).

**Review by mattclarke — APPROVED** [review]
Just the one comment, once that is resolved this can be merged

File: tests/parse_config_update_test.py [hunk 4]
```diff
@@ -170,3 +173,50 @@ def test_parse_streams_skips_stream_info_if_remove_config_and_schema_present_but
...
+    side_effect=flatbuffer_struct.error("Flatbuffer Error"),
+)
+def test_command_is_invalid_on_flatbuffer_struct_error(mock_func):
+    message = serialise_rf5k(
+        UpdateType.ADD,
+        [StreamInfo("test_channel", "f142", "output_topic", Protocol.PVA)],
+    )
+    config_update = parse_config_update(message)
+    assert config_update.command_type == CommandType.INVALID
+
```
[the remark this answers is not in the document]
Comment (kmurica) [reply tests/parse_config_update_test.py:202]: On a second thought, I will just add the test case. Can't possibly hurt. :)

**Pull request merged**